pkg/local_object_storage/shard/mode.go

```diff
@@ -68,3 +69,14 @@ func (s *Shard) GetMode() mode.Mode {
 
 	return s.info.Mode
 }
+
+// UpdateReadonlyMetric updates "readonly" metric in prometheus.
+func (s *Shard) UpdateReadonlyMetric() {
```

Comment:
Why can't we do it in `SetMode`? It is the only place where we change shard mode and we don't need any additional logic on the caller.

Reply:
The idea is to have metric for shard mode right after the startup, not only when shard mode changed somehow.

Reply:
could it be an unexported method that is called on startup and on mode changes?

Reply:
I've removed UpdateReadonlyMetric(), added IsReadOnly() and refactored a bit SetMode and AddShard. How about this?